### PR TITLE
Don't delete user's purchased tickets after user unregisters

### PR DESCRIPTION
**Pull request by rishabhs95**

### CODE REVIEW

File: app/views/conference_registrations/show.html.haml [hunk 1]
```diff
@@ -116,8 +116,12 @@
       -if @registration
         .btn-group-vertical.pull-right
           = link_to 'Edit your Registration', edit_conference_conference_registration_path(@conference.short_title), class: 'btn btn-success', disabled: @conference.end_date < Date.today
-          = link_to 'Unregister', conference_conference_registration_path(@conference.short_title),
-          method: :delete, class: 'btn btn-danger btn-xs', confirm: 'Are you sure you want to unregister?', disabled: @conference.end_date < Date.today
+          - if @tickets.any?
+            = link_to 'Unregister', conference_conference_registration_path(@conference.short_title), method: :delete, class: 'btn btn-danger btn-xs',
+              data: { confirm: "Your ticket purchases won't be deleted. Are you sure you want to unregister?" }, disabled: @conference.end_date < Date.today
```
Comment (differentreality) [inline app/views/conference_registrations/show.html.haml]: I don't think that deleting a ticket purchase makes much sense to an attendee. What we should be saying - and I assume that's what you meant to communicate here - is that there will be no refund.

**Comment by differentreality** [comment]
I am wondering if this is currently possible, after we merged the ticketing PR. It really should not!

**Comment by rishabhs95** [comment]
Currently, if a user unregisters, ticket purchases are not persisted... what we tend to improve in this PR... After this the User's ticket purchases will persist there after they unregister.

File: app/views/conference_registrations/show.html.haml (hunk 1 above)
Comment (rishabhs95) [inline app/views/conference_registrations/show.html.haml]: I think I will update the message to a more appropriate one as ""Your ticket purchases won't be deleted and the money won't be refunded. Are you sure you want to unregister?"

**Comment by rishabhs95** [comment]
Yes, It will be disabled after merging this PR @differentreality

File: app/views/tickets/index.html.haml [hunk 2]
```diff
@@ -37,7 +37,11 @@
             = button_tag(type: 'submit', class: 'btn btn-success btn-lg') do
               Continue
               %i.fa.fa-shopping-cart
-            = link_to 'Cancel registration', conference_conference_registration_path(@conference.short_title), method: :delete, class: 'btn btn-danger btn-sm'
+            - if current_user.ticket_purchases.by_conference(@conference).any?
+              = link_to 'back to registration', conference_conference_registration_path(@conference.short_title), class: 'btn btn-default btn-sm'
```
Comment (differentreality) [inline app/views/tickets/index.html.haml]: Back instead of back (capital B)

File: app/views/tickets/index.html.haml [hunk 3]
```diff
@@ -37,7 +37,11 @@
...
+            -else
```
Comment (differentreality) [inline app/views/tickets/index.html.haml]: Space after -

**Comment by differentreality** [comment]
@rishabhs95 this looks OK! 
Please squash your commits, and rebase.

**Comment by rishabhs95** [comment]
I have updated the PR with squashed commits :)

File: spec/models/registration_spec.rb [hunk 4]
```diff
@@ -91,22 +91,14 @@
     end
   end
 
-  describe '#destroy_purchased_tickets' do
-    it 'destroys purchased tickets if tickets are purchased' do
+  describe 'does not destroy purchased tickets' do
```
Comment (differentreality) [inline spec/models/registration_spec.rb]: I think this is no longer a model test and should be moved to a feature test. @ChrisBr ?

**Comment by differentreality** [comment]
@rishabhs95 can you move the test in features?

**Comment by rishabhs95** [comment]
sure :)

**Comment by differentreality** [comment]
@rishabhs95 nice, thanks! :-) 
Let's squash the commits into 1 and merge this!

**Comment by rishabhs95** [comment]
Squashed :)

**Comment by differentreality** [comment]
👍

**Pull request merged**